internal/app/app.go

```diff
@@ -0,0 +1,73 @@
+package app
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/webdevelop-pro/go-common/configurator"
+	"github.com/webdevelop-pro/go-common/logger"
+	"github.com/webdevelop-pro/migration-service/internal/adapters"
+	"github.com/webdevelop-pro/migration-service/internal/domain/migration"
+)
+
+const pkgName = "migration"
+
+type App struct {
+	log  logger.Logger
+	repo adapters.Repository
+	set  *migration.Set
+	cfg  *Config
+}
+
+func New(c *configurator.Configurator, repo adapters.Repository) *App {
+	cfg := c.New(pkgName, &Config{}, pkgName).(*Config)
+
+	return &App{
+		log:  logger.NewDefaultComponent(pkgName),
+		repo: repo,
+		cfg:  cfg,
+		set:  migration.New(cfg.Dir, repo),
+	}
+}
+
+func (a *App) ApplyAll() error {
+	n, err := a.set.ApplyAll(a.cfg.ForceApply)
+	if err != nil {
+		if err.Error() == "failed to get service version for migration_seed: query failed: ERROR: relation \"migration_service\" does not exist (SQLSTATE 42P01)" {
```

Comment:
it doesn't look good. you can handle pgx error like [this](https://github.com/jackc/pgx/wiki/Error-Handling)

Reply:
or you can use a construction like this `create table if not exists`


```
CREATE TABLE IF NOT EXISTS migration_service (
		id serial NOT NULL PRIMARY KEY,
		name varchar NOT NULL UNIQUE,
		version int NOT NULL DEFAULT 0,
		created_at timestamp with time zone DEFAULT now() NOT NULL,
		UNIQUE (name, version)
	);
	```

Reply:
@darland done